numbat/modules/math/combinatorics.nbt

```diff
@@ -0,0 +1,34 @@
+use core::error
+use core::functions
+use math::transcendental
+
+@name("Factorial")
+@description("The product of the integers 1 through n, also written n!")
+@url("https://en.wikipedia.org/wiki/Factorial")
+fn factorial(n: Scalar) -> Scalar = n!
+
+@name("Falling factorial")
+@description("Equal to n⋅(n-1)⋅…⋅(n-k+2)⋅(n-k+1) (k terms total). If n is an integer, this is the number
+	of k-element permutations from a set of size n. k must always be an integer.")
+@url("https://en.wikipedia.org/wiki/Falling_and_rising_factorials")
+fn falling_factorial(n: Scalar, k: Scalar) -> Scalar =
+	if k < 0 || fract(k) != 0 then
```

Comment:
We have `is_non_negative`. Maybe we should also add `is_integer` and use both functions here (and below)?